packages/core/src/api/nodeConversions/nodeConversions.ts

```diff
@@ -55,14 +79,7 @@ export function inlineContentToNodes(
 
   for (const content of blockContent) {
     if (content.type === "link") {
-      // special case for links, because in ProseMirror, links are marks
-      // in BlockNote, links are a type of inline content
-      const linkMark = schema.marks.link.create({
-        href: content.href,
-      });
-      nodes.push(
-        ...styledTextArrayToNodes(content.content, schema, [linkMark])
-      );
+      nodes.push(...linkToNodes(content, schema));
```

Comment:
I think it might still be a good idea to change the if/else if chain to a lookup map:
```
export function inlineContentToNodes(
  blockContent: PartialInlineContent[],
  schema: Schema
): Node[] {
  let nodes: Node[] = [];
  
  const lookup = {
    "text": (content) => nodes.push(...styledTextArrayToNodes([content], schema)),
    "link": (content) => nodes.push(...linkToNodes(content, schema))
  }

  for (const content of blockContent) {
    if (content.type in lookup) {
      lookup[content.type](content)
    } else {
      throw new UnreachableCaseError(content);
    }
  }
  return nodes;
}

```

Would make it easier to add support for more inline content types in the future.

Reply:
Let's do this only when it becomes relevant. At this point the downside is it would break the typesafe union-extensiveness check (`UnreachableCaseError`)